### PR TITLE
Feature/ Horizontal Bar Visual Chart

### PR DESCRIPTION
**Pull request by olerichard**
# Description

A version of BarVisual, where the bars are horizontal, instead of vertical.
Difference here is minimnal, and setup is the same as BarVisual

Completes: [AB152126](https://dev.azure.com/Equinor/Johan%20Castberg%20-%20Portal/_workitems/edit/152126/)

## Checklist:

-   [x] I have performed a self-review of my own code.
-   [x] I have linked my DevOps task using the AB# tag.
-   [x] My code is easy to read, and comments are added where needed.
-   [ ] My code is covered by tests.

### CODE REVIEW

File: src/packages/Diagrams/src/Visuals/HorizontalBarVisual/Hooks/useHorizontalBarChart.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import { useMemo } from 'react';
+import { BarChartOptions } from '../Types/barVisualOptions';
+import { createSeriesByKeys } from '../Utils/createSeriesByKeys';
+
+interface BarChart {
+    barChartOptions: {
```
Comment (Noggling) [inline src/packages/Diagrams/src/Visuals/HorizontalBarVisual/Hooks/useHorizontalBarChart.ts:6]: her kan du bruke type som react-apexcharts eller apexcharts kan privide så slipper vi å type det opp selv ;)

**Review by Noggling — COMMENTED** [review]
ellers ser dette bra ut ;)

File: src/packages/Diagrams/src/Visuals/HorizontalBarVisual/Hooks/useHorizontalBarChart.ts (hunk 1 above)
Comment (Noggling) [reply src/packages/Diagrams/src/Visuals/HorizontalBarVisual/Hooks/useHorizontalBarChart.ts:6]: `
import { ApexOptions } from 'apexcharts';
import { useMemo } from 'react';
import { BarChartOptions } from '../Types/barVisualOptions';
import { createSeriesByKeys } from '../Utils/createSeriesByKeys';

interface BarChart {
    barChartOptions: ApexOptions;
    series: string[] | { name: string; type: string; data: number[] }[];
}

export function useBarChart<T>(
    data: T[],
    { stacked, nameKey, categoryKey, colors }: BarChartOptions<T>
): BarChart {
    const { series, categories } = useMemo(
        () => createSeriesByKeys(data, 'column', nameKey as string, categoryKey as string),
        [categoryKey, data, nameKey]
    );

    const barChartOptions: ApexOptions = useMemo(
        () => ({
            chart: {`

Comment (Noggling) [reply src/packages/Diagrams/src/Visuals/HorizontalBarVisual/Hooks/useHorizontalBarChart.ts:6]: eksemper på hvordan ;)

Comment (olerichard) [reply src/packages/Diagrams/src/Visuals/HorizontalBarVisual/Hooks/useHorizontalBarChart.ts]: Oppdatert og til å bruk ApexOptions i stedet.

**Review by Gustav-Eikaas — APPROVED** [review]
LGTM

**Pull request merged**